pkg/instrumentation/javaagent.go

```diff
@@ -50,12 +50,18 @@ func injectJavaagent(logger logr.Logger, javaSpec v1alpha1.Java, pod corev1.Pod,
 			logger.Info("Skipping javaagent injection, the container defines JAVA_TOOL_OPTIONS env var value via ValueFrom", "container", container.Name)
 			return pod
 		}
-		container.Env[idx].Value = container.Env[idx].Value + javaJVMArgument
+
+		if IsEnvVarValueInstrumentationMissing(container.Env[idx], javaJVMArgument) {
+			container.Env[idx].Value = container.Env[idx].Value + javaJVMArgument
+		}
+	}
+
+	if IsOtAIVolumeMissing(container.VolumeMounts) {
```

Comment:
Why is this check needed? 

The `IsAutoInstrumentationInjected` is not enough?

Reply:
So `IsOtAIVolumeMissing` and `IsEnvVarValueInstrumentationMissing` were applied before I've added check for pod mutation. When Pod was mutated few times k8s was throwing errors like `duplicated volume name`. In case of env var it happened that e.g. `javaJVMArgument` was added few times to `JAVA_TOOL_OPTIONS`. So I left this checks for safety for other unexpected mutations.

Reply:
I would prefer to remove this and `IsEnvVarValueInstrumentationMissing` as well and rely only on the init container for simplicity. 

We should keep it here only if there is a known scenario which the init container does not cover